### PR TITLE
Decorate a class

### PR DESCRIPTION
**Pull request by franclobo**
#### Interface
- Create a class `Nameable`.
- Implement a method called `correct_name` that will raise a `NotImplementedError`.

#### Turn your Person class to Nameable
- Person class inherits from Nameable
- Create a method `correct_name` implemented, that simply returns the name attribute.

#### Prepare base Decorator
- Class Decorator inherits from Nameable.
- Assign a nameable object from params to an instance variable in the constructor.
- Implement the `correct_name` method that returns the result of the correct_name method of the `@nameable`.

#### Prepare CapitalizeDecorator and TrimmerDecorator
- For the CapitalizeDecorator:
    - Create a class that inherits from the base Decorator class.
    - Implement a method `correct_name` that capitalizes the output of `@nameable.correct_name`.
- For the TrimmerDecorator:
    - Create a class that inherits from the base Decorator class.
    - Implement a method `correct_name` that makes sure that the output of `@nameable.correct_name` has a maximum of 10 characters. If it's longer it should trim the word.

### CODE REVIEW

File: classes/nameable.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class Nameable
+  def correct_name
+    raise NotImplementedError
```
Comment (rustam-tolipov) [inline classes/nameable.rb]: - [x] Kindly add some error text to display. 
 Example:     `raise NotImplementedError, "#{self.class} has not implemented method '#{__method__}'"`

File: classes/trimmer_decorator.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+require_relative './decorator'
+
+class TrimmerDecorator < Decorator
+  def correct_name
+    @nameable.correct_name[0, 10]
```
Comment (rustam-tolipov) [inline classes/trimmer_decorator.rb]:
```suggestion
        @nameable.correct_name.length > 10 ? @nameable.correct_name[0..9] : @nameable.correct_name
```
This is more reliable solution

File: classes/decorator.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
+require_relative './nameable'
+
+class Decorator < Nameable
+  def initialize(nameable)
+    super()
```
Comment (rustam-tolipov) [inline classes/decorator.rb:5]: - [x] You do not need to call `super()`. You can user `attr_accessor :nameable` instead.

**Review by rustam-tolipov — CHANGES_REQUESTED** [review]
Hi @franclobo, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
_Your project is almost complete!_
_No Linters errors_


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: classes/nameable.rb (hunk 1 above)
Comment (franclobo) [reply classes/nameable.rb]: Thank you for being clear with your recommendation.

File: classes/trimmer_decorator.rb (hunk 2 above)
Comment (franclobo) [reply classes/trimmer_decorator.rb]: Thank you it will be helpful when I review my code in the future.

File: classes/decorator.rb (hunk 3 above)
Comment (franclobo) [reply classes/decorator.rb:5]: Thank you, but it generated an offense in Rubocop:

image

**Review by tresorsawasawa — APPROVED** [review]
# STATUS: APPROVE :heavy_check_mark: :heavy_check_mark:  :champagne: 

Hi @franclobo 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 :clap: :clap: :100: 

nodding-yes

## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Good PR description :heavy_check_mark: 
- Your README file is professional :heavy_check_mark: 

## Optional suggestions
_Every comment with the **[OPTIONAL]** prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please ping me [@tresorsawasawa](https://github.com/tresorsawasawa) when you comment so I can receive the notification or use slack with the same name._

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**